ipfs-cluster-service/main.go

```diff
@@ -261,7 +261,82 @@ removal, upgrade state using this command, and restart every peer.
 					Action: func(c *cli.Context) error {
 						err := upgrade()
 						checkErr("upgrading state", err)
-						return err
+						return nil
+					},
+				},
+				{
+					Name:  "export",
+					Usage: "save the IPFS Cluster state to a human readable json file",
+					Description: `
+
+This command reads the current cluster state and saves it as json for 
+human readability and editing.  In general any version of cluster will only
+work with the latest state format, so out of date snapshots will be migrated
+before the export.
+`,
+					Action: func(c *cli.Context) error {
+						err := export()
+						checkErr("exporting state", err)
+						return nil
+					},
+				},
+				{
+					Name:  "import",
+					Usage: "load an IPFS Cluster state from an exported human readable json file",
+					Description: `
+
+This command reads in a json encoding of the state specified in the argument
+storing the state as a persistent snapshot to be loaded as the cluster state
+when the cluster peer is restarted.  Paths are assumed to be relative to the
+cluster base directory.  In the case no matches are found the path is then
+assumed to be absolute.
+`,
+					Action: func(c *cli.Context) error {
+						// Get the importing file path
+						importFile := c.Args().First()
+						if importFile == "" {
+							errNoImportFile := errors.New("No import file specified")
+							checkErr("getting import path", errNoImportFile)
+						}
+						_, clusterCfg, _, _, _, _, _, _, _ := makeConfigs()
+						importFile = filepath.Join(clusterCfg.BaseDir, importFile)
+						if _, err := os.Stat(importFile); os.IsNotExist(err) {
+							importFile, err = filepath.Abs(c.Args().First())
+							checkErr("error setting absolute path for importFile", err)
+						}
+
+						// Attempt to read the file to import
+						r, err := os.Open(importFile)
+						checkErr("reading import file", err)
+						err = stateImport(r)
+						checkErr("importing state", err)
+						return nil
+					},
+				},
+				{
+					Name:  "cleanup",
+					Usage: "cleanup persistent consensus state so cluster can start afresh",
+					Description: `
+This command removes the persistent state that is loaded on startup to determine this peer's view of the
+cluster state.  While it removes the existing state from the load path, one invocation does not permanently remove
+this state from disk.  This command renames cluster's data folder to <data-folder-name>.old.0, and rotates other
+deprecated data folders to <data-folder-name>.old.<n+1>, etc for some rotation factor before permanatly deleting 
+the mth data folder (currently defaults to 5)
+`,
+					Action: func(c *cli.Context) error {
+						_, _, _, _, consensusCfg, _, _, _, _ := makeConfigs()
+
+						// Execution lock
+						err := locker.lock()
+						checkErr("acquiring execution lock", err)
+						defer locker.tryUnlock()
+
+						dataFolder, err := raft.MakeDataFolder(consensusCfg.BaseDir, consensusCfg.DataFolder)
```

Comment:
I think it's not necessary to export/use this. CleanupRaft ends up calling internally to mkdir.